src/optimizer/statistics/expression/propagate_comparison.cpp

```diff
@@ -22,6 +22,19 @@ FilterPropagateResult StatisticsPropagator::PropagateComparison(BaseStatistics &
 	default:
 		return FilterPropagateResult::NO_PRUNING_POSSIBLE;
 	}
+	switch (right.type.InternalType()) {
```

Comment:
Any reason for adding this check? The left and right type should be identical no?